pkg/satellite/validation.go

```diff
@@ -0,0 +1,75 @@
+// Copyright (C) 2018 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package satellite
+
+import (
+	"net/mail"
+	"regexp"
+
+	"github.com/zeebo/errs"
+
+	"storj.io/storj/pkg/utils"
+)
+
+const (
+	passMinLength      = 6
+	passMinNumberCount = 1
+	passMinAZCount     = 1
+)
+
+func addError(list *[]error, err error) {
+	if err == nil {
+		return
+	}
+
+	*list = append(*list, err)
+}
+
+func validateUserInfo(info *UserInfo) error {
+	errs := make([]error, 0)
+
+	addError(&errs, checkEmailAddressSyntax(info.Email))
+	addError(&errs, validateFirstName(info.FirstName))
+	addError(&errs, validatePassword(info.Password))
+
+	return utils.CombineErrors(errs...)
+}
+
+// ErrValidation validation related error class
+var ErrValidation = errs.Class("validation error")
+
+func checkEmailAddressSyntax(email string) error {
+	_, err := mail.ParseAddress(email)
+	return ErrValidation.Wrap(err)
+}
+
+func validateFirstName(name string) error {
+	var errs []error
+
+	if name == "" {
+		errs = append(errs, ErrValidation.New("firstName can't be empty"))
+	}
+
+	return utils.CombineErrors(errs...)
+}
+
+func validatePassword(pass string) error {
+	var errs []error
+
+	if len(pass) < passMinLength {
+		errs = append(errs, ErrValidation.New("password can't be less than %d characters", passMinLength))
+	}
+
+	numReg := regexp.MustCompile("[0-9]+")
+	if len(numReg.FindAllString(pass, passMinNumberCount)) < passMinNumberCount {
+		errs = append(errs, ErrValidation.New("password should contain at least %d numbers", passMinNumberCount))
+	}
+
+	alpReg := regexp.MustCompile("[(?i)a-z]+")
```

Comment:
Non need to recompile the regular expression every time. Also use a plain-old loop for this rather than using a slow regexp.